common/src/stack/pylib/stack/wizard.py

```diff
@@ -186,12 +186,12 @@ def getDevices(self):
 		self.data.devices = devices
 		return devices
 
-	def validateDomain(self, value):
-		s = value.split(".")[0]
-		if s.lower() == "frontend" or s.lower() == "backend":
-			return (False, "Cannot have an appliance name as a hostname", "Hostname Error")
+	def validateDomain(self, domain):
```

Comment:
I appreciate the refactor as this method could definitely be more readable.

Requests:

`domain` is really the contents of the hostname field in the wizard.  This should probably be called `fqdn`, since the domain is the piece after the `split('.')`.

`appliance` then, is really the `hostname`.  We happen to be checking it against a couple known appliance names.

Another (perhaps more pythonic) way to write this would be:

```
if hostname.lower() in ['frontend', 'backend']:
```